service/actions/websocket.js

```diff
@@ -224,7 +224,7 @@ async function handleNewPostEvent(msg, dispatch, getState) {
         }
     ]), getState);
 
-    if (userId === users.currentId || post.channel_id === currentChannelId || isSystemMessage(post)) {
+    if ((userId === users.currentId && !isSystemMessage(post)) || post.channel_id === currentChannelId || shouldIgnorePost(post)) {
```

Comment:
maybe I'm wrong but I beileve those kind of system messages will mark the channel as read because you need to be in that channel